core/src/main/scala/org/apache/spark/api/python/PythonRDD.scala

```diff
@@ -889,21 +891,40 @@ private class PythonAccumulatorParam(@transient private val serverHost: String,
     socket
   }
 
-  override def zero(value: JList[Array[Byte]]): JList[Array[Byte]] = new JArrayList
+  override def reset(): Unit = {
+    this._acc = new JArrayList[Array[Byte]]
+  }
+
+  override def isZero: Boolean = {
+    this._acc.isEmpty
+  }
 
-  override def addInPlace(val1: JList[Array[Byte]], val2: JList[Array[Byte]])
-      : JList[Array[Byte]] = synchronized {
+  override def copy(): PythonAccumulatorV2 = {
+    val newAcc = new PythonAccumulatorV2(serverHost, serverPort)
+    newAcc._acc = this._acc
```

Comment:
So the new accumulator API uses the copy method method inside of the `copyAndReset` which is used inside of `writeReplace` during serialization. More generally you want to ship a "clean" accumulator to the workers rather than something which is already potentially have some accumulated values inside of it.